app/controllers/reports_controller.rb

```diff
@@ -0,0 +1,54 @@
+class ReportsController < ApplicationController
+  before_action :authenticate_user!, only: [:create, :new, :edit, :update, :destroy]
+
+  before_action :set_review, only: [:new, :create]
+  before_action :check_admin, only: [:index, :destroy]
+
+  def index
+    @reports = Report.all
+  end
+
+  def new
+    @report = Report.new
+  end
+
+  def create
+    @report = Report.new(report_params)
+    @review.reported = true
+    @review.save
+    respond_to do |format|
+      if @report.save
+        format.html { redirect_to root_path, notice: 'Report was successfully created.' }
+        format.json { render :show, status: :created, location: @movie }
+      else
+        format.html { render :new }
+        format.json { render json: @report.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    @report = Report.find(params[:id])
+    @review = Review.find(@report.review_id)
```

Comment:
what are you doing here?
will @report.review fetch the same record or not?

Reply:
Yes sir, my bad approach.